src/js/common/schemaform/review/ReviewChapters.jsx

```diff
@@ -0,0 +1,187 @@
+    }
+  }
+
+  scrollToChapter = (chapterKey) => {
+    scroller.scrollTo(`chapter${chapterKey}ScrollElement`, window.VetsGov.scroll || {
+      duration: 500,
+      delay: 2,
+      smooth: true,
+    });
+  }
+
+  handleEdit = (pageKey, editing, index = null) => {
+    const fullPageKey = `${pageKey}${index === null ? '' : index}`;
+    if (editing) {
+      this.props.setViewedPages([fullPageKey]);
+    }
+    this.props.setEditMode(pageKey, editing, index);
+  }
+
+  handleSetData = (...args) => {
+    this.props.setData(...args);
+    if (this.props.handleSetData) {
+      this.props.handleSetData();
+    }
+  }
+
+  render() {
+    const {
+      chapters,
+      form,
+      formContext,
+      setValid,
+      viewedPages
+    } = this.props;
+
+    return (
+      <div className="input-section">
+        <div>
+          {chapters.map(chapter => (
+            <ReviewCollapsibleChapter
+              expandedPages={chapter.expandedPages}
+              chapterFormConfig={chapter.formConfig}
+              chapterKey={chapter.name}
+              form={form}
+              formContext={formContext}
+              key={chapter.name}
+              onEdit={this.handleEdit}
+              open={chapter.open}
+              pageKeys={chapter.pageKeys}
+              setData={(...args) => this.handleSetData(...args)}
+              setValid={setValid}
+              showUnviewedPageWarning={chapter.showUnviewedPageWarning}
+              toggleButtonClicked={() => this.handleToggleChapter(chapter)}
+              uploadFile={this.props.uploadFile}
+              viewedPages={viewedPages}/>
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state, ownProps) {
+  // from ownprops
+  const {
+    formConfig,
+    formContext,
+    pageList
+  } = ownProps;
+
+  // from redux state
+  const form = state.form;
+  const formData = state.form.data;
+  const openChapters = getReviewPageOpenChapters(state);
+  const viewedPages = getViewedPages(state);
+
+  const chapterNames = getActiveChapters(formConfig, formData);
+  const disableSave = formConfig.disableSave;
+  const pagesByChapter = createPageListByChapter(formConfig);
+  const chapters = chapterNames.reduce((chaptersAcc, chapterName) => {
+    const pages = pagesByChapter[chapterName];
+
+    const activePages = getActivePages(pages, formData);
+    const expandedPages = expandArrayPages(activePages, formData);
+    const chapterFormConfig = formConfig.chapters[chapterName];
+    const open = openChapters.includes(chapterName);
+    const pageKeys = getPageKeys(pages, formData);
+    const showUnviewedPageWarning = pageKeys.some(key => !viewedPages.has(key));
+
+    chaptersAcc.push({
+      expandedPages,
+      formConfig: chapterFormConfig,
+      name: chapterName,
+      open,
+      pageKeys,
+      showUnviewedPageWarning
+    });
+
+    return chaptersAcc;
+  }, []);
+
+  return {
+    chapters,
+    disableSave,
```

Comment:
I don't think this is passed anywhere?

Reply:
nope just forgot to remove that from `mapStateToProps`